frontend/package.json

```diff
@@ -144,51 +133,65 @@
   },
   "devDependencies": {
+    "@babel/preset-env": "^7.24.7",
+    "@babel/preset-typescript": "^7.24.7",
     "@fortawesome/fontawesome-common-types": "^6.5.1",
     "@react-pdf/types": "^2.1.0",
-    "@testing-library/dom": "^6.15.0",
-    "@testing-library/jest-dom": "^4.2.4",
-    "@testing-library/react": "^9.3.2",
-    "@testing-library/user-event": "^7.1.2",
-    "@turf/helpers": "^6.5.0",
+    "@testing-library/dom": "^10.3.1",
+    "@testing-library/jest-dom": "^6.1.3",
+    "@testing-library/react": "^16",
+    "@testing-library/user-event": "^14.5.2",
+    "@turf/helpers": "^7",
     "@types/chart.js": "^2.9.21",
     "@types/colormap": "^2.3.1",
     "@types/d3": "^5.7.2",
     "@types/dompurify": "^2.2.3",
     "@types/geojson": "^7946.0.7",
-    "@types/jest": "^24.0.0",
+    "@types/jest": "^29.5.4",
     "@types/lodash": "^4.14.149",
     "@types/marked": "^4.0.2",
-    "@types/node": "^16.0.0",
+    "@types/node": "20.8.8",
     "@types/papaparse": "^5.0.3",
-    "@types/react": "^16.13.0",
+    "@types/react": "^18",
     "@types/react-datepicker": "^2.11.0",
-    "@types/react-dom": "^16.9.0",
+    "@types/react-dom": "^18",
     "@types/react-pdf": "^5.7.2",
     "@types/react-redux": "^7.1.7",
     "@types/react-router-dom": "^5.1.3",
     "@types/redux-mock-store": "^1.0.6",
+    "@typescript-eslint/eslint-plugin": "^6.0.0",
+    "@typescript-eslint/parser": "^6.0.0",
+    "@vitejs/plugin-react-swc": "^3.5.0",
+    "babel-jest": "^29.7.0",
     "cross-env": "^7.0.2",
-    "eslint": "^6.8.0",
-    "eslint-config-airbnb": "^18.0.1",
-    "eslint-config-prettier": "^6.11.0",
-    "eslint-config-react-app": "^5.2.0",
-    "eslint-import-resolver-typescript": "^2.0.0",
-    "eslint-plugin-flowtype": "^3.0",
+    "cross-fetch": "^4.0.0",
+    "dotenv": "^16.4.5",
+    "eslint": "^8.45.0",
+    "eslint-config-airbnb": "^19.0.4",
+    "eslint-config-prettier": "^9.1.0",
+    "eslint-config-react-app": "^7.0.1",
+    "eslint-import-resolver-typescript": "^3.6.1",
+    "eslint-plugin-flowtype": "^8.0.3",
     "eslint-plugin-fp": "^2.3.0",
-    "eslint-plugin-import": "^2.20.1",
-    "eslint-plugin-jsx-a11y": "^6.2.3",
-    "eslint-plugin-prettier": "^3.1.4",
-    "eslint-plugin-react": "^7.19.0",
-    "eslint-plugin-react-hooks": "^1.7",
-    "fs": "^0.0.1-security",
+    "eslint-plugin-import": "^2.29.1",
+    "eslint-plugin-jsx-a11y": "^6.8.0",
+    "eslint-plugin-prettier": "^5.1.3",
+    "eslint-plugin-react": "^7.34.1",
+    "eslint-plugin-react-hooks": "^4.6.2",
+    "eslint-plugin-react-refresh": "^0.4.7",
+    "jest": "^29.6.4",
+    "jest-environment-jsdom": "^29.6.4",
+    "prettier": "^3.2.5",
     "redux-mock-store": "^1.5.4",
     "surge": "^0.23.1",
     "timezone-mock": "^1.3.6",
+    "ts-jest": "^29.1.1",
     "ts-node": "^10.9.2",
-    "typescript": "4.4.4"
+    "typescript": "^5.2.2",
+    "vite": "^5.2.0",
+    "vite-plugin-node-polyfills": "^0.21.0"
   },
   "engines": {
-    "node": "16.x"
+    "node": "20.x"
```

Comment:
@ericboucher do you use [NVM](https://github.com/nvm-sh/nvm)? I find it useful when switching between node packages and always appreciate a [.nvmrc](https://github.com/nvm-sh/nvm?tab=readme-ov-file#nvmrc) file to make this step slightly easier. 

I suggest we add an .nvmrc file that references `20.x`